pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator.go

```diff
@@ -69,6 +69,11 @@ func (mutator *VMsMutator) Mutate(ar *admissionv1.AdmissionReview) *admissionv1.
 	mutator.setDefaultMachineType(&vm, preferenceSpec)
 	mutator.setPreferenceStorageClassName(&vm, preferenceSpec)
 
+	if ar.Request.Operation == admissionv1.Create {
+		// Add foreground finalizer
+		vm.Finalizers = append(vm.Finalizers, v1.VirtualMachineControllerFinalizer)
```

Comment:
I think it would be best to consider the upgrade case where  the VMs already exist. That would also be in line  with the VMI controller which adds the finalizer as one of the first operation. 

This should "auto-fix" existing VMs and for new VMs we add the finalizer before we create the VMI.

What do you think?

Reply:
Thanks for the suggestion and the offline explanation! I pushed the changes to address this! PTAL!